### PR TITLE
More clarity in steps to be followed

### PR DESCRIPTION
**Pull request by PalashBansal**
1. Scilexer.dll to be copied for
2. vcvarsall.bat needs to be executed for
both 'Scilexer.dll with boost' & 'Scilexer.dll without boost'.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,8 +35,12 @@ like to debug Notepad++, but don't have boost.
 ## To build `notepad++.exe`:
 
  1. Open [`PowerEditor\visual.net\notepadPlus.vcxproj`](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/visual.net/notepadPlus.vcxproj)
- 2. Build Notepad++ [like a normal Visual Studio project](https://msdn.microsoft.com/en-us/library/7s88b19e.aspx).
-
+ 2. Copy `SciLexer.dll` from `scintilla\bin\` to the same directory as `notepad++.exe`.
+    - For the `Unicode Release` configuration, the output directory
+      (where `notepad++.exe` is) is `PowerEditor\bin\`.
+    - For the `Unicode Debug` configuration, the output directory
+      (where `notepad++.exe` is) is `PowerEditor\visual.net\Unicode Debug\`.
```
Comment (MAPJe71) [inline README.md]: You don't need the `SciLexer.dll` to be able to build `Notepad++`.

Comment (PalashBansal) [inline README.md]: Yes, not to build but to execute. It was mentioned under heading "To build SciLexer.dll without boost:"

File: README.md [hunk 2]
```diff
@@ -45,11 +49,12 @@ Here are the instructions to build SciLexer.dll (for both 32-bit & 64-bit) for N
...
  1. Download the [Boost source code](http://sourceforge.net/projects/boost/files/boost/1.55.0/).
     v1.55 should be used with VS 2013. Then unzip it. In my case, `boost_1_55_0` is copied in `C:\sources\`
- 2. Go to `scintilla\boostregex\` then run BuildBoost.bat with your boost path.
+ 2. You may need to follow this step- Open command prompt, run vcvarsall.bat (e.g. "C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat").
+ 3. From *same* command prompt, Go to `scintilla\boostregex\` then run BuildBoost.bat with your boost path.
     In my case: `BuildBoost.bat C:\sources\boost_1_55_0`
 	If you are compiling a 64 bit Scintilla under your *VS2013 x64 Native tool command prompt*, add `-x64` flag.
 	In my case: `BuildBoost.bat C:\sources\boost_1_55_0 -x64`
- 3. Go in `scintilla\win32\` then run `nmake -f scintilla.mak`
+ 4. Go in `scintilla\win32\` then run `nmake -f scintilla.mak`
```
Comment (MAPJe71) [inline README.md]: Yes, I can understand that you'd want to extend the instructions in the section "To build `SciLexer.dll` _with_ boost". But why not extend them as done in the next section "To build `SciLexer.dll` _without_ boost"? Only the applicable ones, to be clear.

FYI  you don't need to run `vcvarsall.bat` to be able to build Boost. You need it to be able to build Scintilla (i.e. `SciLexer.dll`) though.

File: README.md [hunk 3]
```diff
@@ -75,12 +80,6 @@ To build SciLexer.dll without PCRE support (for both 32-bit & 64-bit):
     - `nmake NOBOOST=1 -f scintilla.mak`         (normal build)
     - `nmake NOBOOST=1 DEBUG=1 -f scintilla.mak` (debugging build)
 
- 4. Copy `SciLexer.dll` from `scintilla\bin\` to the same directory as `notepad++.exe`.
-    - For the `Unicode Release` configuration, the output directory
-      (where `notepad++.exe` is) is `PowerEditor\bin\`.
-    - For the `Unicode Debug` configuration, the output directory
-      (where `notepad++.exe` is) is `PowerEditor\visual.net\Unicode Debug\`.
-
```
Comment (MAPJe71) [inline README.md]: This copy instruction is required for both the Scintilla build _with_ and _without_ Boost.

**Comment by PalashBansal** [comment]
Yes I understood now that
-> README.md is for building and not for executing,
-> vcvarsall.bat is for building Scintilla and not for Boost,
But as _**vcvarsall.bat**_ and _**copying of SciLexer.dll**_ are only mentioned as a part of "**To build SciLexer.dll without boost:**" ( https://github.com/notepad-plus-plus/notepad-plus-plus#to-build-scilexerdll-without-boost ) heading, It depicts that **step 1(3rd subpoint). Regarding vcvarsall.bat** & **step 4. Regarding copying of SciLexer.dll** are **_only_** part of **building SciLexer without Boost** and not part of **building SciLexer with Boost**.
So, my mistake, point about **copying of SciLexer.dll** must not be a part of "**To build notepad++.exe:**"
But it must also not be a continuation in heading **building SciLexer without Boost**, it must be **_separate point in last, for both build with/without Boost._**
Same thing for vcvarsall.bat.
As a newbie here, I felt problem in building with boost, looking into only **To build SciLexer.dll with boost:**, completely ignoring all what is written in **To build SciLexer.dll without boost:**.

So, Is it ok if I make this change-
untitled
I moved-
**"You may need to follow this step- Open command prompt, run vcvarsall.bat (e.g. "C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat")."**
to just above step-
**"From same command prompt, Go in scintilla\win32\ then run nmake -f scintilla.mak"**.
as vcvarsall.bat is for scintilla and not boost.
And copied-
**"Copy SciLexer.dll from scintilla\bin\ to the same directory as notepad++.exe.
For the Unicode Release configuration, the output directory (where notepad++.exe is) is PowerEditor\bin\.
For the Unicode Debug configuration, the output directory (where notepad++.exe is) is PowerEditor\visual.net\Unicode Debug\."**, lengthier but more understandable IMO.

**Comment by MAPJe71** [comment]
Just apply/commit the changes, it will make it easier to review them.

**Comment by donho** [comment]
The provided instructions is even more confused to me. OTOH, we prefer the PR for the Notepad++ binary.

**Pull request closed without merging**